dotCMS/src/test/java/com/dotcms/content/elasticsearch/ESQueryCacheTest.java

```diff
@@ -0,0 +1,272 @@
+import org.jetbrains.annotations.NotNull;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.business.cache.provider.CacheProviderStats;
+import com.dotmarketing.business.cache.transport.CacheTransport;
+import com.google.common.collect.ImmutableSet;
+
+public class ESQueryCacheTest {
+
+
+    static ESQueryCache cache;
+
+
+    @BeforeClass
+    public static void setup() {
+
+        cache = new  ESQueryCache(admin, true);
+        
+        
+        
+    }
+
+    @Test
+    public void test_hash() {
+
+        String testQuery =RandomStringUtils.randomAscii(200);
+
+        SearchRequest request1 = getSearchRquest(testQuery, true) ;
+        SearchRequest request2 = getSearchRquest(testQuery, true) ;
+        
+
+        assertEquals(cache.hash(request1), cache.hash(request2));
+        
+        
+        SearchRequest request3 = getSearchRquest(testQuery, false) ;
+        
+
+        assertNotEquals(cache.hash(request1), cache.hash(request3));
+    }
+
+    
+    @Test
+    public void test_get_hits_by_query() {
+
+        SearchRequest request = getSearchRequest();
+        SearchHits hits= hits();
+        cache.put(request, hits);
+        
+        assertTrue(cache.get(request).isPresent());
+        
+        assertEquals(hits, cache.get(request).get());
+        
+        SearchRequest request2 = getSearchRequest();
+        assertTrue(!cache.get(request2).isPresent());
+
+    }
+    
+    
+    @Test
+    public void test_get_hits_by_request() {
+
+        final String testQuery =RandomStringUtils.randomAscii(200);
+        SearchHits hits= hits();
+        
+        SearchRequest req1 = getSearchRquest(testQuery, true);
+        SearchRequest req2 = getSearchRquest(testQuery, true);
+        cache.put(req1, hits);
+        
+
+        assert(cache.get(req1).isPresent());
+        
+        assert(cache.get(req2).isPresent());
+        
+        assertEquals(cache.get(req1).get(), cache.get(req2).get());
+
+    }
+    
+    
+    @Test
+    public void test_clear_cache() {
+
+        final String testQuery =RandomStringUtils.randomAscii(200);
+        SearchHits hits= hits();
+        
+        SearchRequest req1 = getSearchRquest(testQuery, true);
+        SearchRequest req2 = getSearchRquest(testQuery, true);
+        cache.put(req1, hits);
+        
+
+        
+        assert(cache.get(req1).isPresent());
+        assert(cache.get(req2).isPresent());
+        
+        cache.clearCache();
+        
+        assert(!cache.get(req1).isPresent());
+        assert(!cache.get(req2).isPresent());
+
+
+    }
+    
+    
+    
+    
+    SearchHits hits() {
+        Random rand = new Random(); 
+        List<SearchHit> hitList = new ArrayList<>();
+        for(int i=0;i< rand.nextInt(10);i++) {
+            Text text = new Text(RandomStringUtils.randomAscii(36));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'text' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5288908)